### PR TITLE
[SENDMAIL][ZIPFLDR][MYDOCS] Fix creation of sendto files

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

- Improve and move `SendTo` files creation code and timing.
- Delete `IDS_DESKLINK` of `shell32`.
- Add `IDS_DESKLINK` into `sendmail`.

JIRA issue: [CORE-16924](https://jira.reactos.org/browse/CORE-16924)

BEFORE:
before
AFTER:
did-it

### CODE REVIEW

File: dll/win32/shell32/shelldesktop/CDesktopBrowser.cpp [hunk 1]
```diff
@@ -483,6 +483,201 @@ HANDLE WINAPI SHCreateDesktop(IShellDesktopTray *Tray)
...
+    HANDLE hFind = FindFirstFileW(szPath, &find);
+    if (hFind == INVALID_HANDLE_VALUE)
+        return FALSE;
+
+    BOOL bFound = FALSE;
+    do
+    {
+        if (wcscmp(find.cFileName, L".") == 0 ||
+            wcscmp(find.cFileName, L"..") == 0 ||
+            _wcsicmp(find.cFileName, L"desktop.ini") == 0)
```
Comment (KyleKatarn31) [inline dll/win32/shell32/shelldesktop/CDesktopBrowser.cpp:583]: No risk of thumbs.db being present ?

File: dll/win32/shell32/shelldesktop/CDesktopBrowser.cpp [hunk 2]
```diff
@@ -483,6 +483,201 @@ HANDLE WINAPI SHCreateDesktop(IShellDesktopTray *Tray)
...
+
+    StringCbCopyW(szSendToFile, sizeof(szSendToFile), pszSendTo);
+    PathAppendW(szSendToFile, PathFindFileNameW(szTarget));
+    StringCbCatW(szSendToFile, sizeof(szSendToFile), L".lnk");
+
+    GetSystemDirectoryW(szShell32, _countof(szShell32));
+    PathAppendW(szShell32, L"shell32.dll");
+
+    HRESULT hr = CreateShellLink(szSendToFile, szTarget, NULL, NULL, NULL,
+                                 szShell32, -IDI_SHELL_MY_DOCUMENTS, NULL);
```
Comment (KyleKatarn31) [inline dll/win32/shell32/shelldesktop/CDesktopBrowser.cpp:611]: Question (maybe stupid) : why "-IDI_SHELL_MY_DOCUMENTS" and not "IDI_SHELL_MY_DOCUMENTS" ?

**Comment by yagoulas** [comment]
This is the wrong place to do it. Leave poor CDesktopBrowser alone.

**Comment by Prince-Kassad** [comment]
XP/2003 has this in the DllRegisterServer of sendmail.dll/zipfldr.dll which is called once during second-stage setup (for the Default User) and then populated for every user when the Default User template is copied. ROS should really do the same...

**Comment by katahiromz** [comment]
```txt
err:(dll/shellext/sendmail/sendmail.cpp:149) CreateShellLink(C:\ReactOS\system32\config\systemprofile\SendTo\My Documents.lnk, C:\ReactOS\system32\config\systemprofile\My Documents) failed!
```
`SHGetSpecialFolderPathW` is malformed in ReactOS setup.

File: dll/win32/shell32/shelldesktop/CDesktopBrowser.cpp (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/shelldesktop/CDesktopBrowser.cpp]: Negative value means icon resource ID rather than icon index.

**Comment by katahiromz** [comment]
@HBelusca User profile is not correctly set in ReactOS Setup.

File: dll/shellext/sendmail/lang/fr-FR.rc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+LANGUAGE LANG_FRENCH, SUBLANG_NEUTRAL
+
+STRINGTABLE
+{
+    IDS_DESKLINK, "Bureau (Créer un raccourci)"
+    IDS_SHORTCUT, "Shortcut to %s"
```
Comment (KyleKatarn31) [inline dll/shellext/sendmail/lang/fr-FR.rc:6]: Should be : "Raccourci vers %s"

Comment (katahiromz) [reply dll/shellext/sendmail/lang/fr-FR.rc:6]: Translation is later.

**Comment by katahiromz** [comment]
mydocs??
mydocs

**Comment by Prince-Kassad** [comment]
@katahiromz mydocs is handled in mydocs.dll in XP/2003. We do not have this DLL yet.

The Setup case probably needs to be specially handled in SHGetSpecialFolderPathW.

**Comment by katahiromz** [comment]
PR #2624 has added system file `mydocs.dll`.

**Comment by katahiromz** [comment]
Ready to go!

File: dll/shellext/mydocs/mydocs.cpp [hunk 4]
```diff
@@ -16,6 +16,43 @@ END_OBJECT_MAP()
 CComModule gModule;
 LONG g_ModuleRefCnt = 0;
 
+static BOOL
+CreateEmptyFile(LPCWSTR pszFile)
+{
+    HANDLE hFile;
+    hFile = CreateFileW(pszFile, GENERIC_WRITE, FILE_SHARE_READ, NULL,
+                        CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+    CloseHandle(hFile);
```
Comment (learn-more) [inline dll/shellext/mydocs/mydocs.cpp:25]: if `hFile == INVALID_HANDLE_VALUE` this might cause an exception (based on debug settings)

Comment (HBelusca) [reply dll/shellext/mydocs/mydocs.cpp:25]: A similar bug is present there too:
https://github.com/reactos/reactos/commit/5538facfdd0edd11038f9dd00bb4a6afec440403#diff-7269d101ea680a5d7880e88eb7a1680cR2224-R2226

Comment (katahiromz) [reply dll/shellext/mydocs/mydocs.cpp]: OK.

**Comment by yagoulas** [comment]
nice. also thanks @Prince-Kassad for the clues

**Pull request merged**